tests/components/onewire/test_config_flow.py

```diff
@@ -0,0 +1,333 @@
+"""Tests for 1-Wire config flow."""
+from pyownet import protocol
+
+from homeassistant.components.onewire.const import (
+    CONF_MOUNT_DIR,
+    CONF_TYPE_OWSERVER,
+    CONF_TYPE_SYSBUS,
+    DEFAULT_OWSERVER_PORT,
+    DEFAULT_SYSBUS_MOUNT_DIR,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TYPE
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from . import setup_onewire_owserver_integration, setup_onewire_sysbus_integration
+
+from tests.async_mock import patch
+
+
+async def test_user_owserver(hass):
+    """Test OWServer user flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert not result["errors"]
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_TYPE: CONF_TYPE_OWSERVER},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "owserver"
+    assert not result["errors"]
+
+    # Invalid server
+    with patch(
+        "homeassistant.components.onewire.onewirehub.protocol.proxy",
+        side_effect=protocol.ConnError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={CONF_HOST: "1.2.3.4", CONF_PORT: 1234},
+        )
+
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "owserver"
+        assert result["errors"] == {"base": "cannot_connect"}
+
+    # Valid server
+    with patch("homeassistant.components.onewire.onewirehub.protocol.proxy",), patch(
+        "homeassistant.components.onewire.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.onewire.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={CONF_HOST: "1.2.3.4", CONF_PORT: 1234},
+        )
+
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)